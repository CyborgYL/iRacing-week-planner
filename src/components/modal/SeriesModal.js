// @flow

import * as React from 'react';
import { useTranslation } from 'react-i18next';
import moment from 'moment';
import classnames from 'classnames';
import Modal from './Modal';
import RaceLength from '../columns/RaceLength';

import allRaces from '../../lib/races';

import styles from '../../styles/main.module.scss';

type Props = {
  onClose: () => void,
  isOpen: boolean,
  ownedTracks: Array<number>,
  seriesId: number,
};

export default function SeriesModal({ onClose, ownedTracks, isOpen, seriesId }: Props) {
  const races = allRaces.filter((race) => race.seriesId === seriesId);
  const { t } = useTranslation();

  const now = moment().utc();

  return (
    <Modal
      isOpen={isOpen}
      onClose={onClose}
      title={t('Tracks for {{series}}', {
        series: t(races[0].series),
      })}
      doneAction={onClose}
    >
      <div className={styles['container-fluid']}>
        <div className={styles['table-responsive']}>
          <table className={styles.table} style={{ fontSize: '0.8em' }}>
            <thead>
              <tr>
                <th>{t('Week')}</th>
                <th>{t('Track')}</th>
                <th>{t('Start')}</th>
                <th>{t('End')}</th>
                <th>{t('Length')}</th>
              </tr>
            </thead>
            <tbody>
              {races.map((race) => {
                const raceWeekEnd = moment(race.startTime).add(race.weekLength);
                const current = now.isBetween(race.startTime, raceWeekEnd);

                const startDate = moment(race.startTime).local().toDate();
                const endDate = moment(race.startTime).local().add(race.weekLength)
                  .subtract(1, 'days')
                  .toDate();

                return (
                  <tr key={race.week} style={current ? { fontWeight: 700 } : {}}>
                    <td>
                      {race.week + 1}
                    </td>
                    <td className={classnames({ [styles.success]: ownedTracks.indexOf(race.trackId) !== -1 })}>
                      {t(race.track)}
                    </td>
                    <td>
                      {t('{{date, YYYY-MM-DD}}', { date: startDate })}
                    </td>
                    <td>
                      {t('{{date, YYYY-MM-DD}}', { date: endDate })}
                    </td>
                    <RaceLength race={race} />
                  </tr>
                );
              })}
            </tbody>
          </table>
        </div>
      </div>
    </Modal>
  );
}
