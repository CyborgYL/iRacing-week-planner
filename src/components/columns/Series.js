// @flow

import * as React from 'react';
import { useTranslation } from 'react-i18next';
import StarIcon from '../icon/StarIcon';
import SeriesModal from '../modal/SeriesModal';
import raceListingStyles from '../styles/raceListing.module.scss';

type Props = {
  race: {
    seriesId: number,
    series: string,
  },
  favouriteSeries: Array<number>,
  ownedTracks: Array<number>,
};

export default function Series({ race, favouriteSeries, ownedTracks }: Props) {
  const [modalOpen, setModalOpen] = React.useState(false);
  const openModal = () => setModalOpen(true);
  const closeModal = () => setModalOpen(false);
  const { t } = useTranslation();

  return (
    <td className={raceListingStyles.clickableCell} onClick={openModal}>
      <div>
        {favouriteSeries.indexOf(race.seriesId) !== -1 ? (
          <StarIcon />
        ) : null}

        <span> </span>

        {t(race.series)}
      </div>

      <SeriesModal isOpen={modalOpen} onClose={closeModal} ownedTracks={ownedTracks} seriesId={race.seriesId} />
    </td>
  );
}
