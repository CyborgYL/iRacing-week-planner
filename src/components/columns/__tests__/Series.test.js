import { describe, test, expect } from '@jest/globals';
import React from 'react';
import { shallow } from 'enzyme';
import SeriesModal from '../../modal/SeriesModal';

import Series from '../Series';

describe('components/columns/Series', () => {
  test('renders correctly', () => {
    const component = shallow(
      <Series
        race={{
          seriesId: 345,
          series: 'Rallycross',
        }}
        favouriteSeries={[123, 763]}
        ownedTracks={[]}
      />,
    );

    expect(component).toMatchSnapshot();

    component.first().simulate('click');

    expect(component).toMatchSnapshot();

    component.find(SeriesModal).first().props().onClose();

    expect(component).toMatchSnapshot();
  });

  test('renders favourite series', () => {
    const component = shallow(
      <Series
        race={{
          seriesId: 345,
          series: 'Rallycross',
        }}
        favouriteSeries={[345, 567]}
        ownedTracks={[123]}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
