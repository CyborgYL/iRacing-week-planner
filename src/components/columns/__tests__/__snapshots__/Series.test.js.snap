// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/columns/Series renders correctly 1`] = `
<td
  className="clickableCell"
  onClick={[Function]}
>
  <div>
    <span>
       
    </span>
    Rallycross
  </div>
  <SeriesModal
    isOpen={false}
    onClose={[Function]}
    ownedTracks={Array []}
    seriesId={345}
  />
</td>
`;

exports[`components/columns/Series renders correctly 2`] = `
<td
  className="clickableCell"
  onClick={[Function]}
>
  <div>
    <span>
       
    </span>
    Rallycross
  </div>
  <SeriesModal
    isOpen={true}
    onClose={[Function]}
    ownedTracks={Array []}
    seriesId={345}
  />
</td>
`;

exports[`components/columns/Series renders correctly 3`] = `
<td
  className="clickableCell"
  onClick={[Function]}
>
  <div>
    <span>
       
    </span>
    Rallycross
  </div>
  <SeriesModal
    isOpen={false}
    onClose={[Function]}
    ownedTracks={Array []}
    seriesId={345}
  />
</td>
`;

exports[`components/columns/Series renders favourite series 1`] = `
<td
  className="clickableCell"
  onClick={[Function]}
>
  <div>
    <StarIcon />
    <span>
       
    </span>
    Rallycross
  </div>
  <SeriesModal
    isOpen={false}
    onClose={[Function]}
    ownedTracks={
      Array [
        123,
      ]
    }
    seriesId={345}
  />
</td>
`;
